cypress/e2e/webTables.cy.js

```diff
@@ -1,7 +1,89 @@
 /// <reference types='cypress' />
+import { generateWorker } from '../support/generate';
 
 describe('Web Tables page', () => {
-  it('', () => {
+  const worker = generateWorker();
 
+  function addWorker() {
+    cy.get('#addNewRecordButton').click();
+
+    cy.get('#firstName').type(worker.firstName);
+    cy.get('#lastName').type(worker.lastName);
+    cy.get('#userEmail').type(worker.email);
+    cy.get('#age').type(worker.age);
+    cy.get('#salary').type(worker.salary);
+    cy.get('#department').type(worker.department);
+
+    cy.get('#submit').click();
+
+    return worker;
+  }
+
+  beforeEach(() => {
+    cy.visit('/');
+  });
+
+  it('should add, edit and delete new worker', () => {
+    addWorker();
+
+    cy.get('#searchBox').type(worker.firstName);
+    cy.get('#edit-record-4').click();
+    cy.get('#userEmail').clear();
+    cy.get('#userEmail').type(`new${worker.email}`);
+    cy.get('#submit').click();
+    cy.contains('div', `new${worker.email}`).should('be.visible');
+
+    cy.get('#delete-record-4').click();
```

Comment:
The selector '#delete-record-4' assumes a specific record ID. Ensure that this ID is dynamically determined based on the worker added to avoid test flakiness.